maven-model-builder/src/main/java/org/apache/maven/model/interpolation/ProblemDetectingValueSource.java

```diff
@@ -60,12 +68,12 @@ public Object getValue( String expression )
 
         if ( value != null && expression.startsWith( bannedPrefix ) )
         {
-            String msg = "The expression ${" + expression + "} is deprecated.";
+            String msg = "The expression ${" + expression + "} is " + status + ".";
```

Comment:
Why do we need this now? I mean `pom.` is gone and such an occurency should be treated as opaque and will not be touched at ll. It will remain as a placeholder in the effective model.

Reply:
These changes were made for the initial idea of making the build fail on `pom.` prefixes. Since we are not failing the build anymore these changes have become unnecessary. Once support for prefixless expressions has been removed this class can be removed entirely. I've reverted the changes.